pandas/core/series.py

```diff
@@ -1177,11 +1180,14 @@ def _get_value(self, label, takeable: bool = False):
             return self.iloc[loc]
 
     def __setitem__(self, key, value) -> None:
-        if not PYPY and using_copy_on_write():
-            if sys.getrefcount(self) <= 3:
+        cow_context = None
+        if not PYPY:
+            if using_copy_on_write() and sys.getrefcount(self) <= 3:
                 warnings.warn(
                     _chained_assignment_msg, ChainedAssignmentError, stacklevel=2
                 )
+            elif warn_copy_on_write() and sys.getrefcount(self) <= 3:
+                cow_context = "chained-assignment"
```

Comment:
I think you need to warn here instead of deferring. This might now warn, if the first op in chained assignment created a copy, e.g.

```
pd.options.mode.copy_on_write = "warn"

df = pd.DataFrame({"a": [1, 2, 3,], "b": 1})

indexer = df.a > 1
df = df.a
df[indexer][0] = 5
```

I think you *could* make a case for not warning here since the current behaviour is a no-op as well, but

- the DataFrame case warns
- I would prefer a warning since cow will also warn

Reply:
So the reason that I didn't yet warn here and deferred it to `Manager.setitem_inplace`, is because I wanted the warning to be more specific, so we can distinguish the case of:

```
df = ..
ser = df["col"]
ser[0] = 0
```
-> here we want to warn that you are setting into a series that is a view of another, and that this change will no longer propagate (in this case no longer update the dataframe)

```
df = ...
df["col"][0] = 0
```

-> here we want to warn that this simply won't work anymore in the future, as this is chained assignment (and the case of it that currently does work and is used widely I think, so I want to have a distinct warning for this one)

Now, while writing this I am not sure anymore why this alone warrants deferring it. I have to check in the tests for the case that made me move this to the lower level (I originally was just warning here, as you suggest, but then changed it)